kadai1/yuonoda/lib/convImages.go

```diff
@@ -0,0 +1,140 @@
+package convImages
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"image"
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var fromFmt = flag.String("from", "jpg", "Specify a format of your original image")
+var toFmt = flag.String("to", "png", "Specify a format which you want to convert the image to")
+var path = flag.String("path", ".", "Path to a directory in which images will be converted recursively")
+
+func getImagePaths(path string, fmt string) ([]string, error) {
+	// ディレクトリ内を再帰的に探索
+	var paths []string
+	err := filepath.Walk(path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+
+			// 対象の拡張子か査定
+			ext := filepath.Ext(path)
+			if ext == "" || ext[1:] != fmt { // TODO jpgの表記
+				return nil
+			}
+			paths = append(paths, path)
+			return nil
+		})
+
+	// エラーがあれば、エラーを返す
+	if err != nil {
+		return []string{}, err
+	}
+	return paths, nil
+}
+
+type img struct {
+	name  string
+	Image image.Image
+}
+
+func (i *img) decode(buf *bytes.Reader, fmt string) error {
+	var err error
+	switch fmt {
+	case "jpg":
+		i.Image, err = jpeg.Decode(buf)
+		break
+	case "png":
+		i.Image, err = png.Decode(buf)
+		break
+	case "gif":
+		i.Image, err = gif.Decode(buf)
+	default:
+		err = errors.New("decode format is incorrect")
+	}
+	return err
+}
+
+func (i *img) encode(buf io.Writer, fmt string) error {
+	// 変換
+	var err error
+	switch fmt {
+	case "png":
+		if err != png.Encode(buf, i.Image) {
+			return err
+		}
+		break
```

Comment:
```suggestion
		return png.Encode(buf, i.Image)
```
- GoはcaseごとにbreakしなくてOK
- エラーを返すだけなら特に比較する必要がない(?)
    - エラーメッセージが必要なら`if err := png.Encode(buf, i.Image); err != nil { return fmt.Errorf("could not encode: %w", err)}`とか......

Reply:
確かに、不要ですね。削除しました！